NullReferenceException in LoggerFactory

This is a minimal reproducution that causes the NullReferenceException:

    new LoggerFactory(Enumerable.Empty<ILoggerProvider>(), (LoggerFilterOptions)null).CreateLogger("Test");

The exception is triggered when the CreateLogger is called but it is related to the LoggerFilterOptions being null.

I'm running in .NET 6 but the library seems to be the Microsoft.Extensions.Logging library is 7.0.0

You should be getting a warning when compiling with nullability check enabled. 

```
warning CS8600: Converting null literal or possible null value to non-nullable type.
```

I think this should be good enough to catch such cases.

--- a/src/libraries/Microsoft.Extensions.Logging/src/StaticFilterOptionsMonitor.cs
+++ b/src/libraries/Microsoft.Extensions.Logging/src/StaticFilterOptionsMonitor.cs
@@ -10,7 +10,7 @@ namespace Microsoft.Extensions.Logging
     {
         public StaticFilterOptionsMonitor(LoggerFilterOptions currentValue)
         {
-            CurrentValue = currentValue;
+            CurrentValue = currentValue ?? throw new ArgumentNullException(nameof(currentValue));
         }
 
         public IDisposable? OnChange(Action<LoggerFilterOptions, string> listener) => null;

